public/app/features/plugins/datasource_srv.ts

```diff
@@ -0,0 +1,152 @@
+import _ from 'lodash';
+import coreModule from 'app/core/core_module';
+import config from 'app/core/config';
+// import pluginLoader from './plugin_loader';
+import { importPluginModule } from './plugin_loader';
+
+export class DatasourceSrv {
+  datasources: any;
+
+  constructor(private $q, private $injector, $rootScope, private templateSrv) {
```

Comment:
looks like you get `$rootScope` but dont set it. Then you use it below in `catch`.
I will add a comment